src/main/java/core/basesyntax/SupplyOperation.java

```diff
@@ -0,0 +1,15 @@
+import java.util.Map;
+
+public class SupplyOperation implements OperationHandler {
+    private final Map<String, Integer> fruitStorage;
+
+    public SupplyOperation(Map<String, Integer> fruitStorage) {
+        this.fruitStorage = fruitStorage;
+    }
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        fruitStorage.put(transaction.getFruit(), 
+                fruitStorage.getOrDefault(transaction.getFruit(), 0) + transaction.getQuantity());
+    }
+}
```

Comment:
The class SupplyOperation should be placed in a corresponding package to adhere to the project structure requirements. Consider moving it to the 'strategy' package as it represents a strategy for handling a supply operation.